carbon-design-system/carbon
# [BUG]: Added onClick in the StructuredListRow

Closes #14037

After discussing with @tay1orjones, we decided to keep the `onChange` behavior and add an `onClick` prop to `StructuredListRow` so that users don't have to wrap StructuredList to call an `onClick` event.

_After approved, I'm going to delete the function in the story._

**New**

- Added `onClick` to the StructuredListRow component. 

#### Testing / Reviewing

- Go to the Selection variant for the StructeredList;
- Click on the row and check the console.
